### PR TITLE
sql: remove `ttl_automatic_column` storage parameter

### LINKED ISSUE #83133: Remove `ttl_automatic_column` storage parameter
The logic around `ttl_automatic_column` is a bit confusing and can be simplified to using `ttl_expire_after` to create and drop the `crdb_internal_expiration` column.

`ttl_expire_after` should be able to be used with or without `ttl_expiration_expression` to customize the logic.

Jira issue: CRDB-16878
Epic CRDB-14592

### PR DESCRIPTION
**Pull request by ecwall**
fixes https://github.com/cockroachdb/cockroach/issues/83133

Release note (sql change): Remove ttl_automatic_column storage param.
The crdb_internal_expiration column is created when ttl_expire_after is set and
removed when ttl_expire_after is reset.

### CODE REVIEW

File: pkg/sql/storageparam/tablestorageparam/table_storage_param.go
Comment (otan) [inline pkg/sql/storageparam/tablestorageparam/table_storage_param.go:165]: you'll need to keep this for backwards compat. send a notice when this is set that this is now a no-op.

Comment (ecwall) [reply pkg/sql/storageparam/tablestorageparam/table_storage_param.go:178]: Added back with a notice.

File: pkg/sql/storageparam/storage_param.go
Comment (otan) [inline pkg/sql/storageparam/storage_param.go:36]: evalCtx already has a `Context`

**Review by otan — APPROVED** [review]
lgtm if you've got blessing from vy :)

**Comment by otan** [comment]
Also worth (in a sep pr) revising the rfc

**Comment by ecwall** [comment]
bors r=otan

**Pull request merged**
